src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -314,8 +314,25 @@ public int Run()
                     compilationRoots.Add(new ILCompiler.DependencyAnalysis.TrimmingDescriptorNode(linkTrimFilePath));
                 }
 
-                if (entrypointModule is { Assembly: EcmaAssembly entryAssembly })
+                // Get TypeMappingEntryAssembly from runtime knobs if specified
+                string typeMappingEntryAssembly = null;
+                foreach (var runtimeKnob in runtimeKnobs)
                 {
+                    var knobAndValue = runtimeKnob.Split('=', 2);
+                    if (knobAndValue.Length == 2 && knobAndValue[0] == "System.Runtime.InteropServices.TypeMappingEntryAssembly")
```

Comment:
Nit: could this be "System.Runtime.InteropServices.TypeMapEntryAssembly" to match TypeMapAssemblyTargetAttribute?